components/datadog/agentparams/params.go

```diff
@@ -117,7 +120,19 @@ func WithAgentConfig(config string) func(*Params) error {
 // WithIntegration adds the configuration for an integration.
 func WithIntegration(folderName string, content string) func(*Params) error {
 	return func(p *Params) error {
-		p.Integrations[folderName] = content
+		confPath := path.Join("conf.d", folderName, "conf.yaml")
+		p.Integrations[confPath] = content
```

Comment:
renaming this to `p.File` might be a bit more correct now, I didn't want to do that until I got some feedback however